pkg/reconciler/pipelinerun/resources/pipelineref.go

```diff
@@ -129,9 +143,24 @@ func resolvePipeline(ctx context.Context, resolver remote.Resolver, name string)
 // older PipelineObject into its v1beta1 equivalent.
 func readRuntimeObjectAsPipeline(ctx context.Context, obj runtime.Object) (v1beta1.PipelineObject, error) {
 	if pipeline, ok := obj.(v1beta1.PipelineObject); ok {
-		pipeline.SetDefaults(ctx)
 		return pipeline, nil
 	}
 
 	return nil, errors.New("resource is not a pipeline")
 }
+
+// verifyResolvedPipeline verifies the resolved pipeline
+func verifyResolvedPipeline(ctx context.Context, pipeline v1beta1.PipelineObject) error {
+	cfg := config.FromContextOrDefaults(ctx)
+	if cfg.FeatureFlags.ResourceVerificationMode != config.SkipResourceVerificationMode && cfg.FeatureFlags.EnableAPIFields == config.AlphaAPIFields {
```

Comment:
The verification might be done with the helper function as in https://github.com/tektoncd/pipeline/blob/main/pkg/apis/version/version_validation.go#L30

Reply:
Given that we already have the ResourceVerificationMode, why are we also checking if alpha API fields are enabled?

Reply:
My personal reason is that I feel more secure to have 2 locks on this new feature. 😂 
The formal reason is that I thought this is a alpha feature so it should be gated. But just realized that this is not related to any APIs, maybe ResourceVerificationMode is enough?